data/block/blockV2_test.go

```diff
@@ -673,6 +886,247 @@ func TestHeaderV2_ValidateHeaderVersion(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestHeaderV2_GetMiniBlockHeadersHashes(t *testing.T) {
+	t.Parallel()
+
+	hash1 := []byte("hash1")
+	hash2 := []byte("hash2")
+
+	hv2 := block.HeaderV2{
+		Header: &block.Header{
+			MiniBlockHeaders: []block.MiniBlockHeader{
+				{Hash: hash1},
+				{Hash: hash2},
+			},
+		},
+	}
+
+	mbhh := hv2.GetMiniBlockHeadersHashes()
+	require.NotNil(t, mbhh)
+	require.Equal(t, 2, len(mbhh))
+	require.Equal(t, hash1, mbhh[0])
+	require.Equal(t, hash2, mbhh[1])
+}
+
+func TestHeaderV2_MapMiniBlockHashesToShards(t *testing.T) {
+	t.Parallel()
+
+	hash1 := []byte("hash1")
+	hash2 := []byte("hash2")
+
+	hash1Shard := core.MetachainShardId
+	hash2Shard := uint32(1)
+
+	hv2 := block.HeaderV2{
+		Header: &block.Header{
+			MiniBlockHeaders: []block.MiniBlockHeader{
+				{Hash: hash1, SenderShardID: hash1Shard},
+				{Hash: hash2, SenderShardID: hash2Shard},
+			},
+		},
+	}
+
+	mbhh := hv2.MapMiniBlockHashesToShards()
+	require.NotNil(t, mbhh)
+	require.Equal(t, 2, len(mbhh))
+	require.Equal(t, hash1Shard, mbhh[string(hash1)])
+	require.Equal(t, hash2Shard, mbhh[string(hash2)])
+}
+
+func TestHeaderV2_IsStartOfEpochBlock(t *testing.T) {
+	t.Parallel()
+
+	hv2 := block.HeaderV2{
+		Header: &block.Header{
+			EpochStartMetaHash: nil,
+		},
+	}
+
+	isStartOfEpoch := hv2.IsStartOfEpochBlock()
+	require.False(t, isStartOfEpoch)
+
+	hv2 = block.HeaderV2{
+		Header: &block.Header{
+			EpochStartMetaHash: []byte("epochStartMetaHash"),
+		},
+	}
+
+	isStartOfEpoch = hv2.IsStartOfEpochBlock()
+	require.True(t, isStartOfEpoch)
+}
+
+func TestHeaderV2_GetBlockBodyTypeInt32(t *testing.T) {
+	t.Parallel()
+
+	bodyType := block.SmartContractResultBlock
+	hv2 := block.HeaderV2{
+		Header: &block.Header{
+			BlockBodyType: bodyType,
+		},
+	}
+
+	bodyTypeInt32 := hv2.GetBlockBodyTypeInt32()
+	require.Equal(t, int32(90), bodyTypeInt32)
```

Comment:
please use the defined constants in the entire PR. 90 -> SmartContractResultBlock

Reply:
done